### DISCUSSION TITLE
Is there a iterative version of `select`?

### DISCUSSION THREAD

**Discussion by astrojhgu** [rust-ndarray/ndarray#1109] [Q&A]
I'd like to iterate an `Array` along one axis over a set of selected indices. However I am not able to find such a method.

I notice that the `fn select(&self, axis, &[Idx])` returns a copy, which is inefficient for read only iterative operation.

Is there such a method fulfill this purpose?

e.g., if there is hypothetical method, I'd like to use it in the way

```
for s in someArray.selected_indices(Axis(0), &[1,2,4,8,16]){
    //some read only operation to s
}
```

**Comment by bluss** [comment]
It's just a matter of calling the right slice or index axis method in each iteration

**Reply by astrojhgu** [reply]
> It's just a matter of calling the right slice or index axis method in each iteration

Let me complement my question. 

I actually need to call an `into_par_iter` on the hypothetical `selected_indices` so that I can zip it with other parallel iterators.

**Reply by bluss** [reply]
I'd say look at what you can do from existing apis
